### PR TITLE
Addded Event and Listener Generation Commands

### PR DESCRIPTION
**Pull request by adelowo**
They follow the same convention as the default commands artisan ships with, e.g the Listener command requires an `--event` option to be present.

### CODE REVIEW

File: src/Commands/GenerateEventCommand.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+	protected $description = 'Generate a new Event Class for the specified module';
+
+	/**
+	 * Create a new command instance.
+	 *
+	 * @return void
+	 */
+	public function __construct()
+	{
+		parent::__construct();
```
Comment (nWidart) [inline src/Commands/GenerateEventCommand.php]: No need for a construct here since it doesn't do anything but calling the parent.

File: src/Commands/stubs/event.stub [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace $NAMESPACE$;
+
+use $DUMMYNAMESPACE$\Event;
```
Comment (nWidart) [inline src/Commands/stubs/event.stub]: Why is this line needed ? What's dummy namespace ? Events by default don't really need to extend anything.

File: src/Commands/stubs/event.stub [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Contracts\Broadcasting\ShouldBroadcast;
```
Comment (nWidart) [inline src/Commands/stubs/event.stub]: You use those 2 classes, but not used in the event class itself.

File: src/Commands/stubs/listener.stub [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        //
+    }
+
+    /**
+     * Handle the event.
+     *
+     * @param $EVENTSHORTENEDNAME$ $event
+     * @return void
+     */
+    public function handle($EVENTSHORTENEDNAME$ $event)
```
Comment (nWidart) [inline src/Commands/stubs/listener.stub]: What's the difference between eventshortenedname & eventname ?

**Comment by nWidart** [comment]
Hello,

Thank you for you contribution.

Besides the couple of comments inline, I have those few notes:

- Please follow psr1/2, meaning using spaces instead of tabs.
- Can you add tests for those 2 commands ? You can view an example of a similar test case [here](https://github.com/nWidart/laravel-modules/blob/master/tests/Commands/MakeRequestCommandTest.php)

File: src/Commands/stubs/listener.stub (hunk 4 above)
Comment (adelowo) [inline src/Commands/stubs/listener.stub]: Eventname is the fully qualified class name (plus namespace).. having to use EVENTSHORTENEDNAME is more like a personal sentiment.. I would fix that though

File: src/Commands/GenerateEventCommand.php (hunk 1 above)
Comment (adelowo) [inline src/Commands/GenerateEventCommand.php]: Got it

File: src/Commands/stubs/event.stub (hunk 2 above)
Comment (adelowo) [inline src/Commands/stubs/event.stub]: I would remove those

File: src/Commands/stubs/event.stub (hunk 3 above)
Comment (adelowo) [inline src/Commands/stubs/event.stub]: The `SerializeModels` class is included in the generated class while the `ShouldBroadCast` is left for the user to implement. Do you want that taken off ?

**Comment by adelowo** [comment]
I had my ide set it (tabs) to 4 spaces, i would just move make it use spaces instead.

File: src/Commands/stubs/event.stub (hunk 2 above)
Comment (nWidart) [inline src/Commands/stubs/event.stub]: Thanks 😄

File: src/Commands/stubs/event.stub (hunk 3 above)
Comment (nWidart) [inline src/Commands/stubs/event.stub]: Hm I think so yea, and people can implement it as needed. Otherwise the IDE like phpstorm will say it's not used and hint to remove it. :)

File: src/Commands/stubs/listener.stub (hunk 4 above)
Comment (nWidart) [inline src/Commands/stubs/listener.stub]: Ahh yes I agree, that makes sense :)

**Comment by nWidart** [comment]
Looks like you forgot to register the commands in the CommandServiceProvider class.

Also in a test file you have a double semicolon `;` and the end of a file.

**Comment by nWidart** [comment]
Tests are still failing. I don't think you registered the commands here https://github.com/nWidart/laravel-modules/blob/master/src/Providers/ConsoleServiceProvider.php#L70

**Comment by nWidart** [comment]
Please squash your commits once you have the tests working.

Also please remove inline comments like 

> // Tests fail because of an undefined variable, event

Thanks

**Comment by adelowo** [comment]
Will do as you've asked..

**Comment by adelowo** [comment]
It's been squashed already and I think the new console commands should be added to the `README` file

**Comment by adelowo** [comment]
Any update on this yet

**Comment by nWidart** [comment]
Thank you for the contribution, will update the docs.

**Comment by nWidart** [comment]
Your tests were failing. :/

**Comment by adelowo** [comment]
Thanks a lot for merging :+1:
